x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/common/H3SphericalUtilTests.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.spatial.common;
+
+import org.apache.lucene.spatial3d.geom.GeoPolygon;
+import org.apache.lucene.spatial3d.geom.LatLonBounds;
+import org.elasticsearch.common.geo.GeoBoundingBox;
+import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.h3.H3;
+import org.elasticsearch.test.ESTestCase;
+
+public class H3SphericalUtilTests extends ESTestCase {
+
+    private static final double DELTA = 1e-8;
+
+    public void testRandomBounds() {
+        GeoBoundingBox boundingBox = new GeoBoundingBox(new GeoPoint(), new GeoPoint());
+        for (int res = 1; res <= H3.MAX_H3_RES; res++) {
+            final long h3 = H3.geoToH3(90, 0, res);
```

Comment:
The test name implies this test should use randomly chosen h3, but it seems to only use the north pole. Perhaps you meant to generate a random h3?

Reply:
Yes, I added randomization. 

Curiously I had to adjust the delta for the bounding box longitude. It seems spatial3d might adjust slightly those values. Still I think should not be a problem for this other way to compute the bounding box.